### ISSUE TITLE
Error loading module with LIT-WORD! name/type (possible AS error)

### ISSUE THREAD

**Issue by rgchris** [metaeducation/ren-c#569]
When DOing a module with a LIT-WORD! value for TYPE and NAME, an error is generated. Looking at where the [source](https://github.com/metaeducation/ren-c/blob/master/src/mezz/base-funcs.r#L945) breaks down, it appears to be related to the application of AS on said LIT-WORD! values:

```
>> do {Rebol [Title: "Foo Bar" Date: 14-Jul-2017 Type: 'module Name: 'foo.bar] 1234}
Module: Foo Bar Version: Date: 14-Jul-2017
** Access Error: series is source or permanently locked, can't modify
** Where: if module catch either either --anonymous-- do
** Near: as word! spec/name ??
** File: ../src/include/sys-series.h
** Line: 507
```

**Comment by hostilefork** [comment]
Well, fixing this aside, what do you think about the protocol.

I don't care if it's lit-words or word-words.  I just want only one to be accepted.  To me, module headers are *data* not *code* and as such it is implicitly understood that they will not be executed.  And as such the least "noisy" choice should stand.  So I would prefer `type: module` to `type: 'module`.

**Comment by rgchris** [comment]
I do prefer the LIT-WORD! form. I know that as an old-school CONSTRUCTed object, LIT-WORD! should not decay to words, it still looks wrong to use words that don't resolve to anything not *litted*.

**Comment by hostilefork** [comment]
> it still looks wrong to use words that don't resolve to anything not litted.

So in JSON-stream style data, do you want this convention too?  Should every wire format obey "code" conventions?

**Comment by rgchris** [comment]
No. My sense is that the header is a dialect that closely resembles an object spec, resolves to an object, but is not evaluated (and also shouldn't contain construction syntax, but that's a separate issue).

**Comment by gchiu** [comment]
https://github.com/metaeducation/ren-c/issues/572
